Jenkinsfile

```diff
@@ -0,0 +1,183 @@
+        }
+        archiveArtifacts artifacts: "${archiveName}-*.tar.gz", onlyIfSuccessful: true
+      }
+    }
+    if (branch != 'master') {
+      return
+    }
+    input 'Push image and tags?'
+    stage('Push image and tags') {
+      def dockerhubApiToken
+      withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: 'docker-hub-credentials',
+            usernameVariable: 'DOCKERHUB_API_USERNAME', passwordVariable: 'DOCKERHUB_API_PASSWORD']]) {
+        dockerImages.each {
+            imageId = it.imageTag
+            def tags = getTags(it.flavor, version)
+            tags.each {
+                OsTools.runSafe(this, "docker tag ${imageId} ${organization}/${dockerHubRepository}:${it}")
+            }
+        }
+        OsTools.runSafe(this, """
+            docker login --username ${env.DOCKERHUB_API_USERNAME} --password ${env.DOCKERHUB_API_PASSWORD}
+            """)
+        OsTools.runSafe(this, "docker push ${organization}/${dockerHubRepository}")
+        
+        response = OsTools.runSafe(this, """
+          curl -X POST https://hub.docker.com/v2/users/login/ \
+            -H 'cache-control: no-cache' -H 'content-type: application/json' \
+            -d '{ "username": "${env.DOCKERHUB_API_USERNAME}", "password": "${env.DOCKERHUB_API_PASSWORD}" }'
+        """)
+        token = readJSON text: response
+        dockerhubApiToken = token.token
+
+        def readme = readFile file: 'README.md', encoding: 'UTF-8'
+        readme = readme.replaceAll("(?s)<!--.*?-->", "")
+        readme = readme.replace("\"", "\\\"")
+        readme = readme.replace("\n", "\\n")
+        response = httpRequest customHeaders: [[name: 'authorization', value: "JWT ${dockerhubApiToken}"]],
+            acceptType: 'APPLICATION_JSON', contentType: 'APPLICATION_JSON', httpMode: 'PATCH',
+            requestBody: "{ \"full_description\": \"${readme}\" }",
+            url: "https://hub.docker.com/v2/repositories/${organization}/${dockerHubRepository}/"
+      }
+    }
+    stage('Push tags') {
+      withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: credentialsId,
+                        usernameVariable: 'GITHUB_API_USERNAME', passwordVariable: 'GITHUB_API_PASSWORD']]) {
+        OsTools.runSafe(this, "git tag ${version}")
+        OsTools.runSafe(this, """
+          git push \
+          https://${env.GITHUB_API_USERNAME}:${env.GITHUB_API_PASSWORD}@github.com/${organization}/${gitHubRepository}.git \
+            ${version}
+        """)
+      }
+      OsTools.runSafe(this, "git tag -d ${version}")
+    }
+  } finally {
+    OsTools.runSafe(this, "docker logout")
+    OsTools.runSafe(this, "docker system prune -a -f")
+    OsTools.runSafe(this, 'git clean -f && git reset --hard origin/master')
+  }
+}
+
+def readVersion() {
+  def content = readFile 'oss/Dockerfile'
+  for (line in content.split('\n')) {
+    if (line.startsWith('ARG NEXUS_VERSION=')) {
+      return line.substring(18).split('-')[0]
```

Comment:
Could use the same regex from updateRepositoryManagerVersion

Reply:
Guess it would be slightly different since it's the shorter version

Reply:
I'm going to pass over this for now as I have a migraine and I do not want to mess with RegEx (lol). Do not disagree with the change, however.